libcontainer/cgroups/fs2/create.go

```diff
@@ -0,0 +1,103 @@
+package fs2
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+func neededControllers(cgroup *configs.Cgroup) []string {
+	var list []string
+
+	if cgroup == nil {
+		return list
+	}
+
+	// list of all available controllers
+	const file = UnifiedMountpoint + "/cgroup.controllers"
```

Comment:
TODO (#2281):  this should be `/sys/fs/cgroup/user.slice/user-1001.slice/cgroup.controllers` in rootless systemd

Reply:
I'm not sure. We're getting the list of controllers available on the host, it does not matter rootless or not. Even if some controllers are not available for the rootless cgroup, we won't error out, so having them in this list is OK.